File: 78-subsets/subsets.cpp
class Solution {
public:
    void solve(int i,int n,vector<int>&temp,vector<int>&nums,vector<vector<int>>&res){
        if(i==n){
            res.push_back(temp);
            return;
        }
        temp.push_back(nums[i]);
        solve(i+1,n,temp,nums,res);
        temp.pop_back();
        solve(i+1,n,temp,nums,res);
        
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        int n=nums.size();
        vector<vector<int>>res;
        vector<int>temp;
        solve(0,n,temp,nums,res);
        return res;
    }
};

File: 78-subsets/subsets.java
class Solution {
    public void solve(int i,int n,int[] nums,List<Integer>temp,List<List<Integer>>ans){
        if(i==n){
            ans.add(new ArrayList<>(temp));
            return;
        }
        temp.add(nums[i]);
        solve(i+1,n,nums,temp,ans);
        temp.remove(temp.size()-1);
        solve(i+1,n,nums,temp,ans);
    }
    public List<List<Integer>> subsets(int[] nums) {
        int n=nums.length;
        List<List<Integer>>ans=new ArrayList<>();
        List<Integer>temp=new ArrayList<>();
        solve(0,n,nums,temp,ans);
        return ans;

    }
}